### ISSUE TITLE
Type of error?

### ISSUE THREAD

**Issue by D219** [Atinux/schema-inspector#54]
I like that library a lot, but I have a problem with the error output. While a human readable message is fine when using it for a pet project, it become a pain to use when you have to map it to a error code.

What I mean is, I can easily know programatically which field have a error, but knowing what raised the error would involve parsing the error message, which is a *horrible* way, since the message could change at any time and break the system.

Would it be possible to add something like "reason" which provide the reason  of the error, like "missing field", "wrong type"? I am willing to add it myself, if you could point me to where the error message is generated.

**Comment by Atinux** [comment]
Hi @D219 

The `error` field might to the work :)
https://github.com/Atinux/schema-inspector#error

**Comment by D219** [comment]
This only replace the error message, or I misunderstand the documentation.

**Comment by Atinux** [comment]
Sorry I did not read carefully your initial message, it's actual a good idea.

I implemented in the version `1.6.7`, you will have a `reason` key which is the `key` where the validation failed.

Example:
```js
var schema = {
	type: 'object',
	properties: {
		lorem: { type: 'string', eq: 'ipsum' },
		dolor: {
			type: 'array',
			items: { type: 'number' }
		}
	}
};
var candidate = {
	lorem: 'not_ipsum',
	dolor: [ 12, 34, 'ERROR', 45, 'INVALID' ]
};
var result = inspector.validate(schema, candidate, function (err, result) {
	console.log(result);
});
```

Result:
```js
{ error: 
   [ { code: null,
       reason: 'eq',
       message: 'must be equal to "ipsum", but is equal to "not_ipsum"',
       property: '@.lorem' },
     { code: null,
       reason: 'type',
       message: 'must be number, but is string',
       property: '@.dolor[2]' },
     { code: null,
       reason: 'type',
       message: 'must be number, but is string',
       property: '@.dolor[4]' } ],
  valid: false,
  format: [Function: format] }
```

If, in your `exec` method, you want to give a reason, it's in the 3rd argument in `this.report(message, code, reason)`
